test/index.test.js

```diff
@@ -743,3 +744,19 @@ test('should restart an app before listening', async (t) => {
   await app.close()
   t.ok(!app.server.listening)
 })
+
+test('should always proxy to the factory wrapped version of app', async (t) => {
+  async function userProvidedFactory () {
```

Comment:
```suggestion
  async function userProvidedFactory (fastify) {
```

You should pass a fastify instance here.

Reply:
Hmm, if you pass the fastify factory as a param (see previous message), your test works on the current version. Not sure that I understand your point. Can you point me to where do you think you receive an app object instead of a proxy wrapper?

Reply:
@ivan-tymoshenko Hi Ivan, actually, you're correct, while the proxy is set to `const app = fastify({ ...newOpts, serverFactory })` in one place, and `newApp = await factory(createApplication, opts, restartOptions)` in another, because the proxy contains a reference to the latest copy of the object and the object is being mutated, they are the same.

Our codebase was missing the fastify factory as a param.